akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/LogCapturing.scala

```diff
@@ -37,7 +37,16 @@ import akka.actor.testkit.typed.internal.CapturingAppender
  */
 final class LogCapturing extends TestRule {
   // eager access of CapturingAppender to fail fast if misconfigured
-  private val capturingAppender = CapturingAppender.get("")
+  private val capturingAppender =
+    try {
+      CapturingAppender.get("")
+    } catch {
+      case iae: IllegalArgumentException if iae.getMessage.contains("it was a [org.slf4j.helpers.NOPLogger]") =>
+        throw new RuntimeException(
+          "SLF could not initialize the logger, this is may be caused by accidentally having the slf4j-api dependency " +
+          "evicted/bumped to 2.2 by transitive dependencies while Akka only supports slf4j-api 1.7",
+          iae)
+    }
```

Comment:
If I understand it correctly, previous slf4j version would return a valid logger when we lookup for `""` and then we would check if its appender is `CapturingAppender`, correct?

It would be better to not rely on this but find another way of lookup the declared appender.

Reply:
Ok maybe looking up for `""` was the how we could find the `root` logger?

Reply:
Ok, I dug a little more in the code to better understand it. 

When looking up for `org.slf4j.Logger.ROOT_LOGGER_NAME`, we get back the `NOPLogger` if using slf4j 2.2. Isn't that strange?

Reply:
If getting slf4j-api 2.2 on path, and a logging provider that doesn't implement 2.2, it is ignored and falls back to that no-op logger, which throws the IAE higher upp in the call chain. I think this is a good place to catch that and ammend some details that may not be obvious.

This dummy lookup makes sure that failure happens early rather than intermingled with test logic where it would possibly seem like an actual test failure.

Reply:
Thanks for the clarification.